### PR TITLE
chore: Remove legacy version check in `react-native.config.js`

### PR DESCRIPTION
**Pull request by lukmccall**
## Description

Removes legacy version check from `react-native.config.js`. 

## Changes

The community CLI is no longer a dependency for React Native, meaning it may not be installed in every project. For example, projects that use Expo won't include the `@react-native-community/cli-*` packages. Consequently, this check will fail, leading to issues with autolinking components.

This version check can be removed safely, as the 9th version of the CLI was released two years ago. 

## Tests

This change was tested using `expo` project on react native `0.76`.

### CODE REVIEW

**Review by emrekaraa — APPROVED** [review]

**Comment by janicduplessis** [comment]
We just released a patch that changes the default to include the config if the cli package doesn't exist, does that work? https://github.com/th3rdwave/react-native-safe-area-context/pull/535

**Comment by vonovak** [comment]
@janicduplessis change from #535 works fine, thank you.

However, there's another interesting point: in many setups, until now, this setting was not considered

```
{
            libraryName: 'safeareacontext',
            componentDescriptors: [
              'RNCSafeAreaProviderComponentDescriptor',
              'RNCSafeAreaViewComponentDescriptor',
            ],
            cmakeListsPath: 'src/main/jni/CMakeLists.txt',
          }
```

and it didn't seem like anything is wrong with how the safe area behaves - instead of using these custom-made files, the default ones that come with React Native were used. I'm wondering if these files are actually necessary?

**Comment by janicduplessis** [comment]
From what I remember it is needed because we use custom component descriptor for SafeAreaView (it has custom state class) so we need to make sure to include it (https://github.com/th3rdwave/react-native-safe-area-context/blob/main/android/src/main/jni/safeareacontext.h#L6). Not sure if autolinking is still somehow able to pick it up without that config now?

**Comment by lukmccall** [comment]
The current code generation does not support custom shadow nodes. This is why you're using the `interfaceOnly` flag [here](https://github.com/th3rdwave/react-native-safe-area-context/blob/e8f873f0d19e2e1119071299cbea6eb70010342f/src/specs/NativeSafeAreaView.ts#L16). This flag ensures that the module's interface will be generated, but it won't be automatically linked. This is the desired behavior in this case, as it allows you to modify the shadow nodes. Additionally, when combined with `react-native.config.js`, it can still be linked.

tl;dr, if you are using a custom component descriptor to implement the shadow node, you cannot remove `componentDescriptors` from `react-native.config.js`

I wouldn't be concerned about backward compatibility with the CLI. The version we are checking against was published two years ago, and there are likely many other reasons why your library won't compile with React Native when paired with CLI version 9. At least, that's my personal opinion.

**Review by janicduplessis — APPROVED** [review]

**Comment by janicduplessis** [comment]
Ok let's just remove it
